src/api/OnmsAuthConfig.ts

```diff
@@ -4,20 +4,29 @@
  */
 export class OnmsAuthConfig {
   /** The password to authenticate with. */
-  public password: string;
+  public password?: string;
 
   /** The username to connect as. */
-  public username: string;
+  public username?: string;
 
   /**
    * Construct an auth configuration object.
    * @constructor
    */
-  constructor(username: string, password: string) {
+  constructor(username?: string, password?: string) {
     this.username = username;
     this.password = password;
   }
 
+  /**
+   * Whether this auth object is the same as another.
+   */
+  public equals(that?: OnmsAuthConfig) {
+    return that &&
+      this.username === that.username &&
+      this.password === that.password;
```

Comment:
I am a fan of adding the `||` or `&&` to the next line, as it is much easier to read:

```
that 
   && this.username === that.username
   && this.password === that.password;
```